src/api.js

```diff
@@ -166,7 +176,14 @@ const Api = (options = {}) => {
         rb,
         hdl: [
           () => parse(request, params, event),
-          paramsOut => handler(paramsOut, context, rb, event)
+          async (paramsOut) => {
+            const result = await handler(paramsOut, context, rb, event);
+            result.fields = (rawAutoPruneFieldsParam !== undefined
+              && paramsOut[rawAutoPruneFieldsParam.name] !== undefined)
+              ? objectPaths.split(paramsOut[rawAutoPruneFieldsParam.name])
+              : null;
```

Comment:
Just so I understand. We could prune here as well? Or is there a drawback?

I think the `fields` feels very forced here. My preference now is to prune here... Sorry, going a bit back and forth here. I should have looked at where / what before you started working on this. Let's do that for the next change...